src/util/Date.h

```diff
@@ -320,17 +320,18 @@ class Date {
 // can be currently represented by the year class [-9999, 9999]. The underlying
 // format is as follows (starting from the most significant bit):
 // - 5 bits that are always zero and ignored by the representation.
-// - 2 bits that store 0 (negative year), 1 (datetime with a year in [-9999,
-// 9999]), or 2 (positive year).
+// - 2 bits that store 1 (negative year), 2 (datetime with a year in [-9999,
+// 9999]), 3 (positive year), or 0 (timezone).
 //   These values are chosen s.t. that the order of the bits is correct.
 // - 57 bits that either encode the `Date`, or a year as a signed integer via
 // the `ad_utility::NBitInteger` class.
 class DateOrLargeYear {
  private:
   // The tags to discriminate between the stored formats.
-  static constexpr uint64_t negativeYear = 0;
-  static constexpr uint64_t datetime = 1;
-  static constexpr uint64_t positiveYear = 2;
+  static constexpr uint64_t timezone = 0;
+  static constexpr uint64_t negativeYear = 1;
+  static constexpr uint64_t datetime = 2;
+  static constexpr uint64_t positiveYear = 3;
```

Comment:
I think if you make the timezon have the value 3 and leave the others, then you don't immediately break all binary indexes, so you should definitely do this.